x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/utils.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { TrainedModelConfigResponse } from '@kbn/ml-plugin/common/types/trained_models';
+
+import { isAlphaNumericOrUnderscore } from '../../../../../../../common/utils/is_alphanumeric_underscore';
+
+import { AddInferencePipelineFormErrors, InferencePipelineConfiguration } from './types';
+
+const NLP_CONFIG_KEYS = [
+  'ner',
+  'classification',
+  'text_classification',
+  'text_embedding',
+  'zero_shot_classification',
+];
+export const isSupportedMLModel = (model: TrainedModelConfigResponse): boolean => {
+  return Object.keys(model.inference_config).some((key) => NLP_CONFIG_KEYS.includes(key));
+};
+
+const RECOMMENDED_FIELDS = ['body', 'body_content', 'title'];
+export const sortSourceFields = (a: string, b: string): number => {
+  const promoteA = RECOMMENDED_FIELDS.includes(a);
+  const promoteB = RECOMMENDED_FIELDS.includes(b);
+  if (promoteA && promoteB) {
+    return RECOMMENDED_FIELDS.indexOf(a) > RECOMMENDED_FIELDS.indexOf(b) ? 1 : -1;
+  } else if (promoteA) {
+    return -1;
+  } else if (promoteB) {
+    return 1;
+  }
+  return a.localeCompare(b);
+};
+
+export const validateInferencePipelineConfiguration = (
+  config: InferencePipelineConfiguration
+): AddInferencePipelineFormErrors | undefined => {
+  const errors: AddInferencePipelineFormErrors = {};
+  if (!isAlphaNumericOrUnderscore(config.pipelineName)) {
+    errors.pipelineName = i18n.translate(
+      'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.configure.invalidPipelineName',
+      {
+        defaultMessage: 'Name must only contain alphanumeric characters and underscores',
+      }
+    );
+  }
+  if (config.destinationField.length > 0 && !isAlphaNumericOrUnderscore(config.destinationField)) {
```

Comment:
I'm not 100% sure this check is required, so advise if it can be removed.

Reply:
I'm starting to think we may not want to do validation on either of these fields, and basically just take any input from the user and let the API request fail if it bad name, but not sure.

Reply:
I'd suggest we check, and if not there, inform (or warn) the user to let them know at least.... @daveyholler , suggestions?

Reply:
I updated the pipeline name validation, and removed the destinationField validation for now.